R/new.R

```diff
@@ -13,6 +13,24 @@ generator_funs$new <- function(...) {
     if (!identical(portable, inherit$portable))
       stop("Sub and superclass must both be portable or non-portable.")
 
+    # If `cloneable` property differs between sub and superclass
+    # - super will override sub if super doesn't allow cloning
+    # - sub will override super if super allows cloning
+    if (!identical(cloneable, inherit$cloneable)) {
+      if (inherit$cloneable) {
+        inherit[["public_methods"]][["clone"]] <- NULL
+      }
+
+      if (!inherit$cloneable) {
+        message(c(
+          "Subclass wants to allow cloning, but superclass has turned it off. ",
+          "Therefore, cloning will also be turned off for subclass."
+        ))
```

Comment:
Placeholder. Can be tweaked further.

Reply:
I suggest including the name of the classes in the message, with something like:

> Superclass XX has cloneable=FALSE, but subclass YY has cloneable=TRUE. A subclass cannot be cloneable when its superclass is not cloneable, so cloning will be disabled for YY.

Also `message()` doesn't need `c()` inside of it.